src/exchange/exchange_client.rs

```diff
@@ -172,6 +175,56 @@ impl ExchangeClient {
         self.post(action, signature, timestamp).await
     }
 
+    pub async fn class_transfer(
+        &self,
+        usdc: f64,
+        to_perp: bool,
+        wallet: Option<&LocalWallet>,
+    ) -> Result<ExchangeResponseStatus> {
+        let usdc = (usdc * 1e6).round() as u64;
+        let wallet = wallet.unwrap_or(&self.wallet);
+
+        let timestamp = next_nonce();
+
+        let action = Actions::SpotUser(SpotUser {
+            class_transfer: ClassTransfer { usdc, to_perp },
+        });
+        let connection_id = action.hash(timestamp, self.vault_address)?;
+        let action = serde_json::to_value(&action).map_err(|e| Error::JsonParse(e.to_string()))?;
+        let is_mainnet = self.http_client.base_url == BaseUrl::Mainnet.get_url();
+        let signature = sign_l1_action(wallet, connection_id, is_mainnet)?;
+
+        self.post(action, signature, timestamp).await
+    }
+
+    pub async fn vault_transfer(
+        &self,
+        is_deposit: bool,
+        usd: String,
+        vault_address: Option<H160>,
+        wallet: Option<&LocalWallet>,
+    ) -> Result<ExchangeResponseStatus> {
+        let vault_address = self
+            .vault_address
+            .or(vault_address)
+            .ok_or_else(|| Error::VaultAddressNotFound)?;
+        let wallet = wallet.unwrap_or(&self.wallet);
+
+        let timestamp = next_nonce();
+
+        let action = Actions::VaultTransfer(VaultTransfer {
+            vault_address,
+            is_deposit,
+            usd,
+        });
+        let connection_id = action.hash(timestamp, self.vault_address)?;
+        let action = serde_json::to_value(&action).map_err(|e| Error::JsonParse(e.to_string()))?;
+        let is_mainnet = self.http_client.base_url == BaseUrl::Mainnet.get_url();
```

Comment:
nit: can you define `HttpClient::is_mainnet`?

Reply:
Done